### PR TITLE
docs: Update CHANGELOG for release

### PR DESCRIPTION
**Pull request by mohanasudhan**
Signed-off-by: Mohan Gandhi <mohgan@amazon.com>

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,25 @@ cfncluster-cookbook CHANGELOG
 
 This file is used to list changes made in each version of the cfncluster cookbook.
 
+1.4.0
+-----
+
+Major new features/updates: 
+
+  - Updated to Amazon Linux 2017.09.1
+  - Updated to Ubuntu 16.04 (server release 20171026.1)
```
Comment (bwbarrett) [inline CHANGELOG.md]: I would say "applied patches to Ubuntu 16.04".  We didn't update to it; we were already there.

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,6 +3,25 @@ cfncluster-cookbook CHANGELOG
...
+  - Updated to Ubuntu 14.04 (server release 20171101)
```
Comment (bwbarrett) [inline CHANGELOG.md]: Same as 16.04

File: CHANGELOG.md [hunk 3]
```diff
@@ -3,6 +3,25 @@ cfncluster-cookbook CHANGELOG
...
+  - Updated to Centos 7.4
+  - Upgraded Centos 6 AMI
+  - Updated to Nvidia driver 384 
+  - Updated to CUDA 9
+  - Updated to latest cfncluster-node 1.4.1 
+
+Bug fixes/minor improvements:
+
+  - Using nvme instead of instance metadata for finding drives
```
Comment (bwbarrett) [inline CHANGELOG.md]: - Added support for NVMe-based instance store.

File: CHANGELOG.md [hunk 4]
```diff
@@ -3,6 +3,25 @@ cfncluster-cookbook CHANGELOG
...
+  - Fixed ganglia plotting issue on ubuntu 
+
```
Comment (bwbarrett) [inline CHANGELOG.md]: - Fixed slow SLURM scaling times on systemd platforms.

**Comment by mohanasudhan** [comment]
Addressed all comments

**Pull request merged**